### PR TITLE
needles and methods for audit module

### PR DESCRIPTION
**Pull request by deepu105**
added new module methods to add menu item to admin menu, its translation 
added method to add angular interceptor
added new methods to rewrite file, replace content, copy templates, html, js with translation logic
added reusable methods for getting installed and all supported languages

added needles for some of the above

refactored some methods from script-base to util.js

amended some error messages for uniformity

removed duplicate declaration in module/index.js

### CODE REVIEW

**Comment by deepu105** [comment]
@jdubois can you plz review this and merge? I need this for my next PR

File: script-base.js [hunk 1]
```diff
@@ -77,7 +76,30 @@ Generator.prototype.addElementToMenu = function (routerName, glyphiconName, enab
...
+        jhipsterUtils.rewriteFile({
+            file: fullPath,
+            needle: 'jhipster-needle-add-element-to-admin-menu',
+            splicable: [
+                    '<li ui-sref-active="active" ><a ui-sref="' + routerName + '" data-toggle="collapse" data-target=".navbar-collapse.in"><span class="glyphicon glyphicon-' + glyphiconName + '"></span>\n' +
+                    '                        &#xA0;<span ' + ( enableTranslation ? 'translate="global.menu.admin.' + routerName + '"':'' ) + '>' + routerName + '</span></a></li>'
+            ]
+        });
+    } catch (e) {
+        console.log(chalk.yellow('\nUnable to find ') + fullPath + chalk.yellow(' or missing required jhipster-needle. Reference to ') + routerName + ' ' + chalk.yellow('not added to menu.\n'));
```
Comment (moifort) [inline script-base.js]: not added to **admin** menu

File: script-base.js [hunk 2]
```diff
@@ -121,11 +143,119 @@ Generator.prototype.addElementTranslationKey = function(key, value, language) {
...
+    try {
+        jhipsterUtils.rewriteFile({
+            file: fullPath,
+            needle: 'jhipster-needle-menu-add-admin-element',
+            splicable: [
+                    '"' + key + '": "' + value + '",'
+            ]
+        });
+    } catch (e) {
+        console.log(chalk.yellow('\nUnable to find ') + fullPath + chalk.yellow(' or missing required jhipster-needle. Reference to ') + language + chalk.yellow(' not added as a new entity in the menu.\n'));
```
Comment (moifort) [inline script-base.js]: not added as a new entity in the **admin** menu

File: script-base.js [hunk 3]
```diff
@@ -409,7 +558,7 @@ Generator.prototype.addSocialConnectionFactory = function (javaDir, importPackag
  *
  */
 Generator.prototype.addMainCSSStyle = function(style, comment) {
-    // Not Working this.useSass -> Undifined
+    // Not Working this.useSass -> Undifined --> this in the modules context would be jhipsterFunc
```
Comment (moifort) [inline script-base.js]: @deepu105  you mean that is the module who give the useSass constant? I need to add a new argument like this:

` function(style, comment, isUseSass)`

File: script-base.js [hunk 4]
```diff
@@ -623,6 +772,107 @@ Generator.prototype.dateFormatForLiquibase = function () {
...
+ * @param {content} content to be written
+ * @param {regex} true if pattern is regex
+ */
+Generator.prototype.replaceContent = function(filePath, pattern, content, regex) {
+    try {
+        jhipsterUtils.replaceContent({
+            file: filePath,
+            pattern: pattern,
+            content: content,
+            regex, regex
```
Comment (moifort) [inline script-base.js]: I don't test and my knowledge in js is quite low but it's normal? it's not to be `regex: regex`?

File: script-base.js (hunk 3 above)
Comment (deepu105) [inline script-base.js]: either that or in your module you need to do `jhipsterFunc.useSass = jhipsterVar.useSass;` coz when you call a method from `jhipsterFunc` your this becomes `jhipsterFunc` and not the running generator context, better add it as param as you said above

Comment (moifort) [inline script-base.js]: Thanks I will make the fix!

File: script-base.js (hunk 4 above)
Comment (deepu105) [inline script-base.js]: its a typo you are correct

**Comment by deepu105** [comment]
updated comments and rebased

**Comment by deepu105** [comment]
awesome

On Mon, Jan 4, 2016 at 10:28 PM, Julien Dubois <notifications@github.com>
wrote:

> Merged #2557 <https://github.com/jhipster/generator-jhipster/pull/2557>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/pull/2557#event-504800573>
> .
>



-- 
thank you,
Deepu.k.s

**Comment by cbornet** [comment]
@deepu105 does the entity hook now work ?

**Comment by jdubois** [comment]
@deepu105 I have travis telling me it's making errors -> very strange, I have this by email but I can't see it in the UI...

**Comment by deepu105** [comment]
@cbornet no i have raised a ticket to yeoman https://github.com/yeoman/yeoman/issues/1582

**Comment by deepu105** [comment]
@jdubois I pushed a commit earlier with a typo it might have failed, then i noticed and commited a correction

**Pull request merged**